gobblin-compaction/src/main/java/org/apache/gobblin/compaction/verify/InputRecordCountHelper.java

```diff
@@ -86,33 +91,98 @@ public long calculateRecordCount (Collection<Path> paths) throws IOException {
   }
 
   /**
+   * Load compaction state file
+   */
+  public State loadState (Path dir) throws IOException {
+    return loadState(this.fs, dir);
+  }
+
+  private static State loadState (FileSystem fs, Path dir) throws IOException {
+    State state = new State();
+    if (fs.exists(new Path(dir, STATE_FILE))) {
+      try (FSDataInputStream inputStream = fs.open(new Path(dir, STATE_FILE))) {
+        state.readFields(inputStream);
+      }
+    }
+    return state;
+  }
+
+  /**
+   * Save compaction state file
+   */
+  public void saveState (Path dir, State state) throws IOException {
+    saveState(this.fs, dir, state);
+  }
+
+  private static void saveState (FileSystem fs, Path dir, State state) throws IOException {
+    Path tmpFile = new Path(dir, STATE_FILE + ".tmp");
+    Path newFile = new Path(dir, STATE_FILE);
+    fs.delete(tmpFile, false);
+    fs.delete(newFile, false);
```

Comment:
Can you move this delete right before the rename to make the window smaller and add a comment that the fs.rename API does not allow atomic rename?